exchanges/stream/websocket.go

```diff
@@ -261,12 +261,14 @@ func (w *Websocket) Connect() error {
 	w.setConnectingStatus(false)
 	w.setInit(true)
 
-	err = w.connectionMonitor()
-	if err != nil {
-		log.Errorf(log.WebsocketMgr,
-			"%s cannot start websocket connection monitor %v",
```

Comment:
Yep, this is what the logs would look like:
```
[WARN]  | WEBSOCKET | 31/05/2023 13:24:28 | Okx websocket has been disconnected. Reason: read tcp 192.168.50.11:52524->104.18.3.151:8443: read: connection reset by peer
[ERROR] | WEBSOCKET | 31/05/2023 13:24:30 | Okx cannot start websocket connection monitor connection monitor is already running
```

I agree that it's the same check that's done inside. It just means Connect will only start the monitor once, or if it is Disabled / Re-Enabled I guess.
This change duplicates it to prevent it being called when it would immediately exit with an error we'd then log.

Alternatives:
* Don't return an error if already running, just silently return early
* Or Catch the error in Connect and ignore it
* Or start connectionMonitor somewhere other that Connect

This seemed the least invasive of the alternatives.
Seeing the reported error often led me to think there was a problem, until I dug into it.
I don't think we want to report an error which isn't a problem at all.

Reply:
Ahh okay, I took your PR to be suggesting that its fixing an error about multiple connectionManagers, but its about highlighting that the error isn't valid for a scenario where one is reconnecting.
Yeah my suggestion was going to be an `errors.Is` inspection, but I'm happy with your check